custom_components/waste_collection_schedule/waste_collection_schedule/source/lipizzanerheimat_at.py

```diff
@@ -0,0 +1,332 @@
+        self.street = street
+        self.town = town
+        self.map_name = map_name
+
+        self.default_headers = {
+            'Host': 'app.jolioo.com',
+            'User-Agent': 'okhttp/3.12.12',
+            'Authorization': self.get_basic_auth_header(USERNAME, PASSWORD),
+            'apikey': APIKEY,
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+
+    @staticmethod
+    def sha1(input_str):
+        return hashlib.sha1(input_str.encode('utf-8')).hexdigest()
+
+    def api_hash(self, endpoint, payload):
+        return self.sha1(f"{APISECRET}@{endpoint}@{payload}")
+
+    @staticmethod
+    def get_basic_auth_header(user, password):
+        credentials = f"{user}:{password}"
+        base64_credentials = base64.b64encode(
+            credentials.encode("utf-8")).decode("utf-8")
+        return f"Basic {base64_credentials}"
+
+    def sha1(self, input_str):
+        return hashlib.sha1(input_str.encode('utf-8')).hexdigest()
+
+    def build_header(self, url, payload):
+        apihash = self.api_hash(url, payload)
+        headers = self.default_headers
+        headers['apihash'] = apihash
+        return headers
+
+    def fetch_communities(self):
+
+        url = 'api/mobile-app/getcommunities'
+        payload = 'language=de'
+
+        headers = self.build_header(url, payload)
+
+        communities = requests.request(
+            "POST", f'{API_URL}/{url}', headers=headers, data=payload).json()['data']
+
+        return next((c['community_id'] for c in communities if c['community_title'] == self.town), None)
+
+    def fetch_calendar_id_by_name(self):
+
+        url = 'api/mobile-app/garbagecalendarindexwithstreets'
+        payload = f'language=de&communities=234|110|110|111|112|241|238|119|113|236|114|237|115|116|117|235&garbage_calendar_id=1'
+
+        headers = self.build_header(url, payload)
+
+        calendarData = requests.request(
+            "POST", f'{API_URL}/{url}', headers=headers, data=payload).json()['data']
+
+        return next((c['garbage_calendar_id']
+                     for c in calendarData['garbage_calendars'] if c['garbage_calendar_title'] == self.map_name), None)
+
+    def fetch_calendar_id(self):
+
+        community_id = self.fetch_communities()
+
+        if (community_id == None):
+            raise Exception("Town/Street combination not found.")
+
+        url = 'api/mobile-app/garbagecalendarindexwithstreets'
+        payload = f'language=de&communities={community_id}&garbage_calendar_id=1'
+
+        headers = self.build_header(url, payload)
+
+        response = requests.request(
+            "POST", f'{API_URL}/{url}', headers=headers, data=payload)
+
+        calendarData = response.json()['data']
+
+        return next((c['garbage_calendars_available'][0]
+                     for c in calendarData['street_structure'] if c['garbage_calendar_street'] == self.street), None)
+
+    def _first_setup(self):
+
+        r = requests.get(API_URL)
```

Comment:
Why this request? `r` is never accessed, and you don't use a session so this does just nothing but an unnecessary web request